gems/pending/util/MiqSockUtil.rb

```diff
@@ -8,6 +8,9 @@ def self.getHostName
   # Return the cannonical name for host in DNS
   def self.getFullyQualifiedDomainName
     Socket.gethostbyname(Socket.gethostname).first
+  rescue
+    $log.error "The appliance hostname could not be resolved. Please make sure to set a valid fully qualified hostname."
```

Comment:
I'm still not sure this is the right place to do this...I feel like it's a caller problem.  This is a general utility and should not even be aware of logging.  In fact, there may be code paths (particular in gems/pending/**/test/*) where the $log is not defined.

Also, similar to my previous comment, "appliance hostname" might not be right here, though I think the other use cases don't exist anymore (like the external SmartProxy), so maybe it's ok.

@gmcculloug can you comment here?